src/probnum/filtsmooth/timeseriesposterior.py

```diff
@@ -36,9 +36,30 @@ class TimeSeriesPosterior(abc.ABC):
         Posterior random variables.
     """
 
-    def __init__(self, locations: np.ndarray, states: np.ndarray) -> None:
-        self.locations = np.asarray(locations)
-        self.states = _randomvariablelist._RandomVariableList(states)
+    def __init__(
+        self,
+        locations: Optional[Iterable[FloatArgType]] = None,
+        states: Optional[Iterable[randvars.RandomVariable]] = None,
+    ) -> None:
+        self._locations = list(locations) if locations is not None else []
+        self._states = list(states) if states is not None else []
```

Comment:
@pnkraemer What do you think of this? This way, we save ourselves an additional `initialize` method that might add unnecessary complexity

Reply:
yes, I am fine with that. When I mentioned the `initialise` methods, I meant one in the filter (not in the posterior). See also my other comment :)